packages/@aws-cdk/aws-apigateway/lib/integrations/stepfunctions.ts

```diff
@@ -0,0 +1,195 @@
+    },
+  ];
+
+  const integResponse = [
+    {
+      statusCode: '200',
+      responseTemplates: {
+        'application/json': `#set($inputRoot = $input.path('$'))
+                #if($input.path('$.status').toString().equals("FAILED"))
+                    #set($context.responseOverride.status = 500)
+                    { 
+                      "error": "$input.path('$.error')",
+                      "cause": "$input.path('$.cause')"
+                    }
+                #else
+                    $input.path('$.output')
+                #end`,
+      },
+    },
+    ...errorResponse,
+  ];
+
+  return integResponse;
+}
+
+function requestTemplates(stateMachine: sfn.IStateMachine, includeRequestContext: boolean | undefined) {
+  const templateStr = templateString(stateMachine, includeRequestContext);
+
+  const requestTemplate: { [contentType:string] : string } =
+    {
+      'application/json': templateStr,
+    };
+
+  return requestTemplate;
+}
+
+function templateString(stateMachine: sfn.IStateMachine, includeRequestContext: boolean | undefined): string {
+  let templateStr: string;
+  const requestContextStr = requestContext();
+
+  const search = '"';
+  const replaceWith = '\\"';
+  if (typeof includeRequestContext === 'boolean' && includeRequestContext === true) {
+    templateStr = `
+    #set($allParams = $input.params())
+    {
+      "input": "{${(requestContextStr.split(search).join(replaceWith))}}",
+      "stateMachineArn": "${stateMachine.stateMachineArn}"
+    }`;
+  } else {
+    templateStr = `
+        #set($inputRoot = $input.path('$')) {
+            "input": "$util.escapeJavaScript($input.json('$'))",
+            "stateMachineArn": "${stateMachine.stateMachineArn}"
+          }`;
+  }
+  return templateStr;
+}
+
+function requestContext(): string {
+  const executionInput: ExecutionInput = new ExecutionInputBuilder('$util.escapeJavaScript($input.json(\'$\'))')
+    .withContext('"requestContext": {')
+    .withAccountId('"$context.identity.accountId"')
+    .withApiId('"$context.apiId"')
+    .withApiKey('"$context.identity.apiKey"')
+    .withAuthorizerPrincipalId('"$context.authorizer.principalId"')
+    .withCaller('"$context.identity.caller"')
+    .withCognitoAuthenticationProvider('"$context.identity.cognitoAuthenticationProvider"')
+    .withCognitoAuthenticationType('"$context.identity.cognitoAuthenticationType"')
+    .withCognitoIdentityId('"$context.identity.cognitoIdentityId"')
+    .withCognitoIdentityPoolId('"$context.identity.cognitoIdentityPoolId"')
+    .withHttpMethod('"$context.httpMethod"')
+    .withStage('"$context.stage"')
+    .withSourceIp('"$context.identity.sourceIp"')
+    .withUser('"$context.identity.user"')
+    .withUserAgent('"$context.identity.userAgent"')
+    .withUserArn('"$context.identity.userArn"')
+    .withRequestId('"$context.requestId"')
+    .withResourceId('"$context.resourceId"')
+    .withResourcePath('"$context.resourcePath"')
+    .withEnd('}')
+    .create();
```

Comment:
Why do we need to call `withApiKey()` with the value `"$context.identity.apiKey"`? Are there other valid values that `withApiKey()` be called with?

This also looks all weird `withContext('"requestContext": {')` and `.withEnd('}')`.

This API does not look right.

Reply:
`withApiKey()` with the context variable shows that the request context string will include the `apiKey`.

Reply:
There was no difference between the two. I have changed the code to avoid confusion.